pkg/sql/drop_type.go

```diff
@@ -61,6 +62,16 @@ func (p *planner) DropType(ctx context.Context, n *tree.DropType) (planNode, err
 			)
 		}
 
+		hasOwnership, err := p.HasOwnership(ctx, typeDesc)
```

Comment:
nit -- it might be a good idea to put these two lines into a function that just returns an error like `canModifyType`